homeassistant/components/keenetic_ndms2/__init__.py

```diff
@@ -1 +1,100 @@
 """The keenetic_ndms2 component."""
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    CONF_CONSIDER_HOME,
+    CONF_INCLUDE_ARP,
+    CONF_INCLUDE_ASSOCIATED,
+    CONF_INTERFACE,
+    CONF_TRY_HOTSPOT,
+    DEFAULT_CONSIDER_HOME,
+    DEFAULT_INTERFACE,
+    DEFAULT_NAME,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TELNET_PORT,
+    DOMAIN,
+)
+from .router import KeeneticRouter
+
+KEENETIC_SCHEMA = vol.All(
+    vol.Schema(
+        {
+            vol.Required(CONF_HOST): cv.string,
+            vol.Required(CONF_USERNAME): cv.string,
+            vol.Required(CONF_PASSWORD): cv.string,
+            vol.Optional(CONF_INTERFACE, default=DEFAULT_INTERFACE): cv.string,
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(CONF_PORT, default=DEFAULT_TELNET_PORT): cv.port,
+            vol.Optional(CONF_TRY_HOTSPOT, default=True): cv.boolean,
+            vol.Optional(CONF_INCLUDE_ARP, default=True): cv.boolean,
+            vol.Optional(CONF_INCLUDE_ASSOCIATED, default=True): cv.boolean,
+            vol.Optional(
+                CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+            ): cv.time_period,
+            vol.Optional(
+                CONF_CONSIDER_HOME, default=DEFAULT_CONSIDER_HOME
+            ): cv.time_period,
```

Comment:
These are new options, we don't allow changes to the YAML configuration.
See ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

Since this change moves YAML from a platform configuration level to an integration level as well, my suggestion would be to remove it.